diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -123,27 +128,33 @@ class BlockStructureBraces : Rule("block-structure") {
         if (thenNode != null && thenNode.hasChildOfType(LBRACE)) {
             checkOpenBraceOnSameLine(node, THEN, configuration)
             checkCloseBrace(thenNode, configuration)
+            if (hasElseBranch) {
+                val allMiddleNode = listOf(node.findChildByType(THEN)!!.treeNext)
+                checkMidBrace(allMiddleNode)
+            }
         }
         if (hasElseBranch && elseNode!!.elementType != IF && elseNode.hasChildOfType(LBRACE)) {
             checkOpenBraceOnSameLine(node, ELSE, configuration)
-            val allMiddleNode =  node.findAllNodesWithSpecificType(THEN).map { it.treeNext }
-            checkMidBrace(allMiddleNode)
             checkCloseBrace(elseNode, configuration)
         }
     }
 
     private fun checkOpenBraceOnSameLine(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
         if (!configuration.openBrace) return
         val braceSpace = node.findChildByType(beforeType)?.treePrev
-        if (braceSpace == null || braceSpace.elementType != WHITE_SPACE || braceSpace.text.contains("\n".toRegex()))
+        if (braceSpace == null || checkBraceNode(braceSpace, true))
             BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace",
                     (braceSpace ?: node).startOffset) {
-
+                if (braceSpace == null || braceSpace.elementType != WHITE_SPACE) {
+                    node.addChild(PsiWhiteSpaceImpl(" "), node.findChildByType(beforeType))
+                } else {
+                    (braceSpace as LeafPsiElement).replaceWithText(" ")
+                }
             }
         checkOpenBraceEndLine(node, beforeType)
     }
 
-    private fun checkOpenBraceEndLine(node: ASTNode, beforeType: IElementType){
+    private fun checkOpenBraceEndLine(node: ASTNode, beforeType: IElementType) {
         val newNode = when (node.elementType) {
             IF -> {
                 when (beforeType) {
```

Comment:
When you call `findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext` I think it it safe to assume that `BLOCK` always has `LBRACE` and add `!!` there.